OpenRA.Mods.Common/AI/BaseBuilder.cs

```diff
@@ -55,7 +114,11 @@ public void Tick()
 				if (TickQueue(queue))
 					active = true;
 
-			waitTicks = active ? ai.Info.StructureProductionActiveDelay : ai.Info.StructureProductionInactiveDelay;
+			// Add a random factor so not every AI produces at the same tick early in the game.
+			// Minimum should not be negative as delays in HackyAI could be zero.
+			var randomFactor = world.SharedRandom.Next(0, ai.Info.StructureProductionRandomBonusDelay);
```

Comment:
Don't use shared random here in future patches. See https://github.com/OpenRA/OpenRA/pull/8870.